### PR TITLE
Glayout LLM + Docs + Notebooks

### PR DESCRIPTION
**Pull request by alibillalhammoud**
- Documentation has been updated to include links to Glayout notebooks
- Added the Via notebook
- All notebooks have been fixed (they broke after we refactored the Glayout package)

Glayout LLM v1
We are getting decent results with the 7b model. It is highly versatile on 2-4 transistor circuits. Current model config:
- 8 bit quantization
- Computing loss on prompt completions only
- Context is included in prompts
- padding with [PAD] token
- Current training dataset: 32 unique circuits, 54 examples
- Current evaluation dataset: 2 unique circuits (currently being expanded)

### CODE REVIEW

File: openfasoc/generators/glayout/glayout/llm/rag_data/CommonDrain.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Common Drain Amplifier
+The purpose of a common drain amplifier, aka a source follower, is to provide buffering. It has high input impedance, low output impedance, and a voltage gain close to unity.
```
Comment (chetanyagoyal) [inline openfasoc/generators/glayout/glayout/llm/rag_data/CommonDrain.md:2]: maybe provide a separate example of a source follower being used in buffering. The LLM might place just a common drain stage as a buffer, which isn't the most ideal

File: openfasoc/generators/glayout/glayout/llm/rag_data/CurrentMirror.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Current Mirror
+A current mirror is a circuit designed to copy a current
```
Comment (chetanyagoyal) [inline openfasoc/generators/glayout/glayout/llm/rag_data/CurrentMirror.md:2]: > designed to copy a current

copy or boost a current

File: openfasoc/generators/glayout/glayout/llm/rag_data/StrongArmLatch.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Strong Arm Latch
+A strong arm latch is a comparator (which means it compares two input voltages). If one voltage is higher it outputs a high voltage and if one of them is lower then it output a low voltage.
```
Comment (chetanyagoyal) [inline openfasoc/generators/glayout/glayout/llm/rag_data/StrongArmLatch.md:2]: this doesn't really make sense, might cause training problems

File: openfasoc/generators/glayout/glayout/llm/rag_data/inverter.md [hunk 4]
```diff
@@ -1,35 +1,5 @@
...
-
-Power Dissipation is a negative side effect of that inverters have. Its typically caused in two instances, static and dynamic. Dynamic power dissipation occurs when switching occurs an a short circuit briefly happens. Static power dissipation happens when the circuit is in steady state.
-
-Noise margins are the amount of noise voltage that can be fed into the input voltage without affecting the output.
\ No newline at end of file
+An Inverter is an electronic circuit which inverts the input. For example, if given a high input voltage it will produce a low output voltage. If given a low input voltage it will produce a high output voltage.
+The pmos is typically placed above the nmos.
+The pmos is above the nmos and it's source is tied to vdd. The nmos source is tied to ground. The nmos drain and pmos drain are routed together. The node connecting the nmos and pmos drain is the output node. Additonally, the nmos gate and pmos gate are connected. The node connecting the nmos and pmos gate is the input node.
```
Comment (chetanyagoyal) [inline openfasoc/generators/glayout/glayout/llm/rag_data/inverter.md:4]: can merge these two lines

**Review by chetanyagoyal — APPROVED** [review]

**Review by msaligane — CHANGES_REQUESTED** [review]
@alibillalhammoud @chetanyagoyal We want these convos to be in the llm folder?

File: openfasoc/generators/glayout/glayout/flow/components/CrossCoupledInverters_cell.py [hunk 5]
```diff
@@ -3,7 +3,7 @@
 # Apache License
```
Comment (msaligane) [inline openfasoc/generators/glayout/glayout/flow/components/CrossCoupledInverters_cell.py:3]: Should we have a header with a little more details?

Comment (msaligane) [reply openfasoc/generators/glayout/glayout/flow/components/CrossCoupledInverters_cell.py:3]: and comments..

**Comment by msaligane** [comment]
Please add my comments as issues to fix later.

**Pull request merged**